### DISCUSSION TITLE
Modifying php.ini values + supervisord for queues

### DISCUSSION THREAD

**Discussion by marcfowler** [coollabsio/coolify#2156] [Q&A] [answered]
Hey all - I'm trying out Coolify and love it so far but need some help on two questions I'm stuck on.

I'm deploying a simple Laravel application and nixpacks is reading my `composer.json` file appropriately and installing the relevant packages for my extensions etc from the require block:

```
"require": {
    "php": "^8.2",
    "ext-exif": "*",
    "ext-intl": "*",
    "ext-sockets": "*"
}
```

This works great but I'm stuck on two points and hoping someone can help.

## Question 1: modifying PHP configuration
I cannot figure out how to modify `php.ini` values such as `upload_max_filesize` etc.

After digging around into nixpacks, it seems like the only way is to modify the nixpacks start cmd to pull in a custom `php-fpm.conf` file when it starts up php-fpm, but I can't find any examples of someone doing that, and that feels like it can't be correct for something that has to be so common!

## Question 2: using supervisord to work Laravel queues
When using something like Laravel Forge, they let you configure a daemon really easily, using supervisord under the hood to run the simple `php artisan queue:work` command.

I've deployed my application, but how do I configure the worker to run as well?

**Comment by marcfowler** [comment, marked as answer]
I've been able to answer my own question here and I've [written a detailed post on how I've customised](https://frontier.sh/posts/20240409-laravel-queues-coolify/) the `nixpacks.toml` file to extend the default PHP provider, pass in `php.ini` values through nginx, and also start supervisor workers to process queues. Hopefully this is useful to others searching!

**Reply by vitalybaev** [reply]
Unfortunately, the page doesn't exist anymore

**Reply by marcfowler** [reply]
Hey @vitalybaev - sorry! Updated link here which includes supervisord running PHP too: https://frontier.sh/posts/20240510-phpfpm-nginx/

**Comment by Nathanjms** [comment]
Hi @marcfowler - thanks for posting this, it was really helpful to help me figure out how to set up a config with Laravel Horizon (similarly to your queues). I've made an example repo for reference [here](https://github.com/Nathanjms/laravel-coolify-example). 

It seems to be working, but I'm unsure about the opcache bit to be honest...

One thing - it may have just been me, but one thing I noticed when I ran yours directly is that my server CPU usage climbed and then stayed permantly on max (200% as 2 cores). I'm unsure if this was something I was doing wrong, but thought it may be worth mentioning in case it helps you spot anything!

Thanks again!
